pkg/controller/knicluster/knicluster_controller.go

```diff
@@ -276,11 +325,32 @@ func (r *ReconcileKNICluster) Reconcile(request reconcile.Request) (reconcile.Re
 	} {
 		err = f(instance, reqLogger)
 		if err != nil {
+			status.SetStatusCondition(&instance.Status.Conditions, kniv1alpha1.Condition{
+				Type:    kniv1alpha1.ConditionDegraded,
+				Status:  corev1.ConditionTrue,
+				Reason:  "ReconcileFailed",
+				Message: fmt.Sprintf("Failed reconcilitation %v", err),
+			})
+
+			_ = r.client.Status().Update(context.TODO(), instance)
```

Comment:
ignored the error here as we have already failed and don't want the err passed up to be about updating the status.